app/src/main/java/android/example/courses/CourseDetailFragment.kt

```diff
@@ -5,16 +5,59 @@ import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.TextView
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
 
 class CourseDetailFragment : Fragment() {
+    private var db = FirebaseFirestore.getInstance()
+    private var auth = FirebaseAuth.getInstance()
+    private var selectedCourse = ""
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        val courseId = arguments?.get("courseId")
-        /* Load the details of the course given by that ID */
-
+        selectedCourse = arguments?.get("courseName") as String
         return inflater.inflate(R.layout.fragment_course_detail, container, false)
     }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+//        val loadingScreen = view.findViewById<FrameLayout>(R.id.progress_overlay)
```

Comment:
@radumarin88 Here I tried to add a loading screen that would show up when the data is loading (It didn't seem to work - the screen remained blank after the data was loaded)